apps/payments/next/app/[locale]/[offeringId]/[interval]/new/page.tsx

```diff
@@ -31,6 +31,12 @@ function getRedirectToUrl(
       ? 'upgrade'
       : 'checkout';
 
+  if (searchParams.coupon && pageType === CartEligibilityStatus.UPGRADE) {
+    const updatedSearchParams = { ...searchParams };
+    delete updatedSearchParams['coupon'];
+    searchParams = updatedSearchParams;
```

Comment:
suggestion (non-blocking): Could this be replaced with `delete searchParams.coupon`?